sandbox/lua/encoders/schema_influx_write.lua

```diff
@@ -0,0 +1,319 @@
+config option.  It can also map any Heka message fields as tags in the
+request sent to the InfluxDB API, using the `tag_fields` config option.
+Tags, timestamp/precision, database, and retention policy are defined once
+for all points submitted in the API call.  All dynamic fields in the Heka
+message are converted to separate series in the `points` array that is
+submitted to InfluxDB.
+
+Config:
+
+- name_prefix (string, optional, default "name")
+    String to use as the `name` key's prefix value in the generated JSON.
+    Supports interpolation of field values from the processed message, using
+    `%{fieldname}`. Any `fieldname` values of "Type", "Payload", "Hostname",
+    "Pid", "Logger", "Severity", or "EnvVersion" will be extracted from the
+    the base message schema, any other values will be assumed to refer to a
+    dynamic message field. Only the first value of the first instance of a
+    dynamic message field can be used for name name interpolation. If the
+    dynamic field doesn't exist, the uninterpolated value will be left in the
+    name. Note that it is not possible to interpolate either the
+    "Timestamp" or the "Uuid" message fields into the name, those
+    values will be interpreted as referring to dynamic message fields.
+
+- database (string, required, default "mydb")
+    The InfluxDB database to store the metrics in.
+
+- retention_policy (string, optional, default "")
+    The InfluxDB retentionPoilcy value for the database in which the
+    metrics are being sent to.  Only relevant when post_v09_api is true.
+
+- skip_fields (string, optional, default nil)
+    Space delimited set of fields that should *not* be included in the
+    InfluxDB records being generated. Any fieldname values of "Type",
+    "Payload", "Hostname", "Pid", "Logger", "Severity", or "EnvVersion" will
+    be assumed to refer to the corresponding field from the base message
+    schema. Any other values will be assumed to refer to a dynamic message
+    field. The magic value "**all_base**" can be used to exclude base fields
+    from being mapped to the event altogether (useful if you don't want to
+    use tags and embed them in the name_prefix instead).
+
+- tag_fields (string, optional, default "**all_base**")
+    Take fields defined and adds them as tags of the point(s) sent to
+    InfluxDB for the message.  The magic values "**all**" and "**all_base**"
+    are used to map all fields (including base) to tags and only base fields
+    to tags, respectively.  If those magic values aren't used, then only those
+    fields defined will map to tags of the points sent to InfluxDB.
+
+- timestamp_precision (string, optional, default "n")
+    Specify the timestamp precision that you want the event sent with.  The
+    default is to use nanoseconds by dividing the Heka message timestamp
+    by 1000000, but this math can be altered by specifying one of the precision
```

Comment:
This would be more readable with separators (i.e. "1,000,000") or, even better, as 1e6.